File: src/main/java/com/LilG/utils/SizedArray.java
package com.LilG.utils;

import java.util.LinkedList;

public class SizedArray<T> extends LinkedList<T> {
    private int maxSize;

    public SizedArray() {
		this(20);
	}

    public SizedArray(int size) {
        super();
        maxSize = size;
    }

    public void setMaxSize(int size) {
        maxSize = size;
		clean();
	}

    @Override
    public boolean add(T object) {
		clean();
		return super.add(object);
	}

    public T get() {
        if (size() == 0) {
            return null;
        }
        return super.get(size() - 1);
    }

	private void clean() {
		//If the array is too big, remove elements until it's the right size.
		while (size() >= maxSize) {
			remove(0);
		}
	}
}

File: src/main/java/com/LilG/utils/SizedArray.kt
package com.LilG.utils

import java.util.*

class SizedArray<T> @JvmOverloads constructor(private var maxSize: Int = 5) : LinkedList<T>() {

    fun setMaxSize(size: Int) {
        maxSize = size
    }

    override fun add(`object`: T): Boolean {
        //If the array is too big, remove elements until it's the right size.
        while (size >= maxSize) {
            removeAt(0)
        }
        return super.add(`object`)
    }

    fun get(): T? {
        if (size == 0) {
            return null
        }
        return super.get(size - 1)
    }
}